src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -660,10 +660,9 @@ internal Collection<FunctionDescriptor> GetFunctionDescriptors(IEnumerable<Funct
             Collection<FunctionDescriptor> functionDescriptors = new Collection<FunctionDescriptor>();
             var httpFunctions = new Dictionary<string, HttpTriggerAttribute>();
 
-            if (!Utility.IsSingleLanguage(functions, _currentRuntimelanguage))
+            if (!_scriptHostEnvironment.IsDevelopment() && !Utility.IsSingleLanguage(functions, _currentRuntimelanguage))
             {
-                _logger.LogError($"Found functions with more than one language. Select a language for your function app by specifying {LanguageWorkerConstants.FunctionWorkerRuntimeSettingName} AppSetting");
-                return functionDescriptors;
+                throw new HostInitializationException($"Found functions with more than one language. Select a language for your function app by specifying {LanguageWorkerConstants.FunctionWorkerRuntimeSettingName} AppSetting");
```

Comment:
LGTM. Assuming you will be adding a test that verifies throwing HostInitializationException